modules/loggers/mod_log_config.c

```diff
@@ -1165,16 +1180,28 @@ static int config_log_transaction(request_rec *r, config_log_state *cls,
     }
 
     for (i = 0; i < format->nelts; ++i) {
-        strs[i] = process_item(r, orig, &items[i]);
-        len += strl[i] = strlen(strs[i]);
+        lfd->portions[i] = process_item(r, orig, &items[i]);
+
+        /* if we don't have a custom log formatter escape all values unconditionally,
+         * else let the custom log formatter handle the escaping
+         */
+        if (!cls->log_formatter) {
```

Comment:
We don't want empty parts here, so possibly:
```
    if (*lfd->portions[i]) {
        lfd->portions[i] = ap_escape_logitem(r->pool, lfd->portions[i])
    }
    else {
        lfd->portions[i] = "\"\"";
    }
```
?

Reply:
Maybe even NULL should be handled here, like:
```
    if (!lfd->portions[i]) {
        lfd->portions[i] = "-";
    }
    else if (!*lfd->portions[i]) {
        lfd->portions[i] = "\"\"";
    }
    else {
        lfd->portions[i] = ap_escape_logitem(r->pool, lfd->portions[i])
    }
```
This would let the formatter handle it (using the `null` value in json for instance).